src/front/app/features/challenge/challenge.reducer.js

```diff
@@ -12,8 +14,12 @@ const reducer = (state = initialState, action = {}) => {
         return upsertChallenge(action.challenge, state);
     case RESET_STATE:
         return initialState;
-    case GET_USER:
+    case USER_RECEIVED:
         return upsertCurrentUser(action.user, state);
+    case ACCEPTED_RECEIVED:
+        return state.setIn(['currentUser', 'acceptedChallenges'], fromJS(action.challenges));
+    case WATCHLIST_RECEIVED:
+        return state.setIn(['currentUser', 'watchList'], fromJS(action.challenges));
     case ADDED_TO_WATCHLIST:
         return state.set('isWatched', true);
```

Comment:
@wips  I understand what is .set method for, 
I don't understand why we need watch list, accepted challenges on challenge details page